test/solution-template-validation-tests/test/mainTemplateTests.js

```diff
@@ -109,18 +103,33 @@ describe('template files - ', () => {
     });
 
     describe('resources tests - ', () => {
-        // Not required pending Jianping's PR
-        it('resourceGroup().location must not be used in the solution template except as a defaultValue for the location parameter', () => {
+        describe('resource location format tests and resourceGroup().location tests - ', () => {
+            var expectedLocation1 = '[parameters(\'location\')]';
+            var expectedLocation2 = '[variables(\'location\')]';
             templateFileJSONObjects.forEach(templateJSONObject => {
                 var templateObject = templateJSONObject.value;
                 templateObject.should.have.property('resources');
-                var locationString = '[resourceGroup().location]';
-                var message = 'in file:' + templateJSONObject.filename + ' should NOT have location set to [resourceGroup().location]';
-                Object.keys(templateObject.resources).forEach(resource => {
-                    var val = templateObject.resources[resource];
-                    if (val.location) {
-                        val.location.should.withMessage(getErrorMessage(val, templateJSONObject.filepath, message)).not.be.eql(locationString);
+                var resources = Object.keys(templateObject.resources).map(function(key) {
+                    return templateObject.resources[key];
+                });
+                // each resource location should be "location": "[parameters('location')]" or ""[variables('location')]""
+                it.each(resources, 'resource named %s resource location should be location: [parameters(\'location\')] or [variables(\'location\')]', ['name'], function(element, next) {
+                    var message = 'in json files should have location set to [parameters(\'location\')] or [variables(\'location\')]';
+                    if (element.location) {
+                        var key = element.location.split(' ').join('');
+                        var locationMap = {};
+                        locationMap[key] = 1;
+                        locationMap.should.withMessage(getErrorMessage(element, templateJSONObject.filepath, message)).have.any.keys(expectedLocation1, expectedLocation2);
                     }
+                    next();
+                });
+                // resourceGroup().location should NOT be present anywhere in template, EXCEPT as a defaultValue
+                it.each(resources, 'resource named %s resourceGroup().location must not be used in the solution template except as a defaultValue for the location parameter', ['name'], function(element, next) {
```

Comment:
resourceGroup().location must NOT be be used in the template for resource %s, except as a default value for the location parameter